// @shoutem/ui/html/components/SimpleHtml.js
import PropTypes from 'prop-types';
import { Dimensions, Linking } from 'react-native';
import React, { PureComponent } from 'react';
import HTML from 'react-native-render-html';
import { View } from '@shoutem/ui';
import { connectStyle } from '@shoutem/theme';

class SimpleHtml extends PureComponent {
  static propTypes = {
    body: PropTypes.string,
    style: PropTypes.object,
    customHandleLinkPress: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.onLinkPress = this.onLinkPress.bind(this);
  }

  onLinkPress(evt, href) {
    const { customHandleLinkPress } = this.props;

    return customHandleLinkPress ? customHandleLinkPress(href) : Linking.openURL(href);
  }

  render() {
    const { style, body } = this.props;
    const { a, ul, ol, img } = style;
    const paddingValue = style.container.paddingLeft * 2;
    const maxWidth = Dimensions.get('window').width - paddingValue;
    const htmlProps = {
      html: body,
      imagesMaxWidth: maxWidth,
      staticContentMaxWidth: maxWidth,
      tagsStyles: { a, ul, ol, img },
      onLinkPress: this.onLinkPress,
    };

    return (
      <View style={style.container}>
        <HTML
          {...htmlProps}
        />
      </View>
    );
  }
}

export default connectStyle('shoutem.ui.SimpleHtml')(SimpleHtml);
